packages/react/src/SelectPanel/SelectPanel.tsx

```diff
@@ -347,13 +406,32 @@ export function SelectPanel({
     return <T extends React.HTMLAttributes<HTMLElement>>(props: T) => {
       return renderAnchor({
         ...props,
-        children: selectedItems.length ? selectedItems.map(item => item.text).join(', ') : placeholder,
+        children: selectedItems.length
+          ? selectedItems
+              .sort(sortFn ?? compareByKey(sortKey, sortDirection))
+              .map(item => item.text)
+              .join(', ')
+          : placeholder,
       })
     }
-  }, [placeholder, renderAnchor, selected])
+  }, [placeholder, renderAnchor, selected, sortDirection, sortFn, sortKey])
 
-  const itemsToRender = useMemo(() => {
-    return items.map(item => {
+  useEffect(() => {
+    if (open) {
+      resetSort()
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [open])
```

Comment:
We could potentially handled this this like in: https://react.dev/learn/you-might-not-need-an-effect#adjusting-some-state-when-a-prop-changes to avoid having to use an effect here. Another idea is to call `resetSort()` when `onClose()` is called (or whatever appropriate prop it would be for when it open/closes)